iceoryx_dds/include/iceoryx_dds/gateway/dds_to_iox.hpp

```diff
@@ -38,13 +37,30 @@ class DDS2IceoryxGateway : public gateway_t
 
   public:
     DDS2IceoryxGateway() noexcept;
+
+    /// @brief Set DDS to iceoryx gateway object
+    /// @param[in] channelFactory the channel factory to propagate data across
     DDS2IceoryxGateway(ChannelFactory channelFactory) noexcept;
+
+    /// @brief Load the provided gateway configuration; a channel is setup if required (no error check on channel setup)
+    /// @param[in] config the gateway configuration to load
     void loadConfiguration(const config::GatewayConfig& config) noexcept;
+
+    /// @brief Discover messages coming from DDS.
+    /// @param[in] msg the discovery message.
     void discover(const capro::CaproMessage& msg) noexcept;
+
+    /// @brief forward data between the two terminals of the channel used by the implementation.
+    /// @param[in] channel the channel to propagate data across.
     void forward(const channel_t& channel) noexcept;
 
   private:
     void* m_reservedChunk = nullptr;
+
+    /// @brief Setup the channel for the given service
+    /// @param[in] service service description of the service to create a channel for
+    /// @param[in] publisherOptions the publisher options with historyCapacity and queueCapacity
+    /// @return an expected containing a copy of the added channel, otherwise an error
```

Comment:
usually we don't document private methods with doxygen. It clutters the public API documentation

Reply:
done